components/src/core/ToolbarMenu/ToolbarMenu.tsx

```diff
@@ -0,0 +1,281 @@
+                            onClick(e);
+                            closeMenu();
+                        };
+                    }
+                }
+            }
+        }
+    }, [menuGroups]);
+
+    const getMenu = useCallback(() => {
+        if (menu && Boolean(anchorEl)) {
+            return React.cloneElement(menu, {
+                anchorEl: anchorEl,
+                open: Boolean(anchorEl),
+                onClose: closeMenu,
+                ...menu.props,
+            });
+        }
+        if (menuGroups && Boolean(anchorEl)) {
+            return (
+                <Menu
+                    getContentAnchorEl={null}
+                    anchorOrigin={{ vertical: 'bottom', horizontal: rtl ? 'right' : 'left' }}
+                    transformOrigin={{ vertical: 'top', horizontal: rtl ? 'right' : 'left' }}
+                    anchorEl={anchorEl}
+                    open={Boolean(anchorEl)}
+                    onClose={(): void => closeMenu()}
+                    {...MenuProps}
+                    MenuListProps={{ style: { padding: 0 } }}
+                >
+                    {!menu &&
+                        menuGroups.map((group: ToolbarMenuGroup1, index: number) => (
+                            <div className={defaultClasses.navGroups} key={index}>
+                                <DrawerNavGroup
+                                    divider={false}
+                                    hidePadding={true}
+                                    itemIconColor={group.iconColor}
+                                    itemFontColor={group.fontColor}
+                                    title={group.title}
+                                    items={group.items.map(
+                                        (item: ToolbarMenuItem1, itemIndex: number): NavItem =>
+                                            Object.assign({ itemID: itemIndex.toString() }, item)
+                                    )}
+                                />
+                            </div>
+                        ))}
+                </Menu>
+            );
+        }
+    }, [menuGroups, menu, anchorEl, MenuProps, defaultClasses]);
+
+    return (
+        <span
+            ref={ref}
+            className={clsx(defaultClasses.root, classes.root, menuGroups || menu ? defaultClasses.cursorPointer : '')}
+            data-test={'wrapper'}
+            {...otherSpanProps}
+        >
+            <Typography
+                ref={anchor}
+                aria-haspopup="true"
+                className={clsx(defaultClasses.labelContent, classes.labelContent)}
+                color={'inherit'}
+                component={'span'}
+                onClick={(): void => {
+                    openMenu(anchor.current);
+                }}
+            >
+                {icon && (
+                    <span className={clsx(defaultClasses.icon, classes.icon)} data-test={'icon'}>
+                        {icon}
+                    </span>
```

Comment:
This could possibly also throw some DOM validation errors, depending on what gets passed in. Since we're inside of a Span element, there might be limitations on what's allowed inside it (e.g., you can't put an H1 inside of a P tag...there might be similar restrictions for span). I'm wondering if we should move the icon and the dropdown arrow outside of the Typography element, and move the onClick handler to the root. I think we might be able to eliminate one of the wrappers here and simplify the class overrides.